# installs
## core stuff
```
sudo apt-get update
sudo apt-get -y upgrade
sudo apt-get -y install tmux git curl gcc make g++ unzip default-jre libdatetime-perl libxml-simple-perl libdigest-md5-perl bioperl
```

## roary (pangenome stuff)
```
sudo apt-get -y install bedtools cd-hit ncbi-blast+ mcl parallel cpanminus prank mafft fasttree
sudo cpanm -f Bio::Roary
```

## prokka dependencies
### hmmer-3
mkdir hmmer
cd hmmer/
wget http://eddylab.org/software/hmmer3/3.1b2/hmmer-3.1b2-linux-intel-x86_64.tar.gz
tar xvzf hmmer-3.1b2-linux-intel-x86_64.tar.gz
cd hmmer-3.1b2-linux-intel-x86_64.tar.gz
sudo ./configure
make
sudo make install

### signalP
Go to the website's [download page](http://www.cbs.dtu.dk/cgi-bin/nph-sw_request?signalp) and enter all your information. Enter a university (.edu) email address. Click the link from the email generated by your request and copy the .tar.gz file to your server. From there, follow their instructions outlined in the `signalp-4.1.readme` file once you've decompressed the file archive. 

## prokka itself
cd $HOME
git clone https://github.com/tseemann/prokka.git
cd prokka/bin/
./prokka --setupdb

# actually run prokka
I had to modify the headers of the home-made fasta files because they initially contained pipes (which an earlier version of Prokka didn't like), so I used spaces. But then I ran a first pass of Prokka and it failed because it stopped after the first space character which was the same for all the fasta headers... so I then substituted the spaces with hyphens... It was these hyphenated strings in the headers that were used for the following prokka command.  

Create a shell script as follows with all your fasta files within that directory:
```
#!/bin/bash
cd /home/ubuntu/bruce/fastas
SAMPLE_LIST=`ls *.fasta | sed 's/.fasta//' | sort -u`
for SAMPLE in $SAMPLE_LIST
do
	prokka \
	--outdir ${SAMPLE} \
	--force \
	--prefix ${SAMPLE} \
	--genus Brucella \
	--cpus 0 \
	${SAMPLE}.fasta
done
```


# actually run Roary
See more details about [install](https://github.com/sanger-pathogens/Roary/blob/master/README.md) and [executing](https://github.com/microgenomics/tutorials/blob/master/pangenome.md) scripts. Ran their generic command first:
```
roary -f ./demo -e -n -v ./gff/*.gff
```


